backend/metadata_writer/src/metadata_writer.py

```diff
@@ -83,11 +84,21 @@ def output_name_to_argo(name: str) -> str:
     import re
     return re.sub('-+', '-', re.sub('[^-0-9a-z]+', '-', name.lower())).strip('-')
 
+def is_s3_endpoint(endpoint: str) -> bool:
+    return re.search('^.*s3.*amazonaws.com.*$', endpoint)
+
+def get_object_store_provider(endpoint: str) -> bool:
+    if is_s3_endpoint(endpoint):
```

Comment:
Should we also add Google Cloud Storage endpoint detection? Probably `storage.googleapis.com`.

Reply:
But then I guess Frontend will have to support it.
Maybe add it later.

Reply:
I can help add this.